### PR TITLE
feat: [PIE-3879]: Refactor GitRemoteDetails component

### PR DESCRIPTION
**Pull request by depanshu-dhiman**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/40-gitsync/components/gitSyncRepoForm/RepoBranchSelect.tsx [hunk 1]
```diff
@@ -126,7 +126,12 @@ const RepoBranchSelect: React.FC<RepoBranchSelectProps> = props => {
           <Text>{getString('common.fetchingBranches').concat('...')}</Text>
         </Layout.Horizontal>
       ) : null}
-      <Dialog isOpen={isOpen} enforceFocus={false} title={getString('gitsync.branchFetchFailed')} onClose={close}>
+      <Dialog
```
Comment (sunnykesh-harness) [inline src/modules/40-gitsync/components/gitSyncRepoForm/RepoBranchSelect.tsx:129]: Extra white space.

File: src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx [hunk 2]
```diff
@@ -5,15 +5,11 @@
...
+import React from 'react'
+import { Icon, Text, SelectOption } from '@wings-software/uicore'
 import { PopoverInteractionKind, Position } from '@blueprintjs/core'
-import { useParams } from 'react-router-dom'
 import cx from 'classnames'
-import { isEmpty, noop } from 'lodash-es'
-import { useGetListOfBranchesByRefConnectorV2 } from 'services/cd-ng'
-import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
-import { getBranchSelectOptions } from '../RepoBranchSelectV2/RepoBranchSelectV2'
+import RepoBranchSelectV2 from '../RepoBranchSelectV2/RepoBranchSelectV2'
```
Comment (sunnykesh-harness) [inline src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx:12]: Please make all other components import as complete path.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx [hunk 3]
```diff
@@ -123,34 +74,22 @@ const GitRemoteDetails = ({
...
-            remoteBranch: {
-              label: branch === defaultBranch || !branch ? (branchText ? `${branchText} (default)` : '') : branch,
-              value: branch
-            }
+        <RepoBranchSelectV2
+          name="remoteBranch"
+          noLabel={true}
+          connectorIdentifierRef={connectorRef}
+          repoName={repoName}
+          onChange={(selected: SelectOption): void => {
```
Comment (sunnykesh-harness) [inline src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx:82]: extra white space

File: src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx [hunk 4]
```diff
@@ -123,34 +74,22 @@ const GitRemoteDetails = ({
...
+            onBranchChange?.({
+              branch: selected.value as string
+            })
           }}
-        >
```
Comment (sunnykesh-harness) [inline src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx:135]: Plz do corresponding CSS clean up too.

File: src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx (hunk 2 above)
Comment (depanshu-dhiman) [reply src/modules/10-common/components/GitRemoteDetails/GitRemoteDetails.tsx:12]: done

File: src/modules/10-common/components/SaveToGitFormV2/SaveToGitFormV2.tsx [hunk 5]
```diff
@@ -124,6 +124,7 @@ const SaveToGitFormV2: React.FC<ModalConfigureProps & SaveToGitFormV2Props> = pr
             }
           }}
           selectedValue={targetBranch}
+          showErrorInModal
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/SaveToGitFormV2/SaveToGitFormV2.tsx:127]: Can we do this only when create PR checkbox is checked ?

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**